subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -843,6 +843,11 @@ static void wl_clear(void)
 	for (int i = 0; i < WL_SIZE; i++) {
 		wl[i].taken = 0U;
 	}
+
+	for (int i = 0; i < CONFIG_BT_CTLR_RL_SIZE; i++) {
```

Comment:
Delete this new code. Instead add `rl[j].wl = 0U;` after line 895.

Reply:
At this point j >= ARRAY_SIZE(rl)?
How come you do not want to remove all 'whitelist refs' in the resolve list when, clearing whitelist?

Reply:
You are right, my fault.

Use lines 908-913 at line 844. This way the for-loop is not needed.

Reply:
I would actually prefer to keep the for-loop, as it does the same thing, is less code and 'easier to read'.

Reply:
You are adding a new loop with a different iteration count, and wl_clear being in the ll_init call path, introduces Bluetooth initialization time penalty.

Reply:
OK, I considered this of lesser complexity, but I guess that is not always true. I applied your modification.